src/main/java/de/zalando/aruha/nakadi/service/subscription/StreamingContext.java

```diff
@@ -0,0 +1,160 @@
+package de.zalando.aruha.nakadi.service.subscription;
+
+import de.zalando.aruha.nakadi.service.subscription.model.Partition;
+import de.zalando.aruha.nakadi.service.subscription.model.Session;
+import de.zalando.aruha.nakadi.service.subscription.state.CleanupState;
+import de.zalando.aruha.nakadi.service.subscription.state.StartingState;
+import de.zalando.aruha.nakadi.service.subscription.state.State;
+import de.zalando.aruha.nakadi.service.subscription.state.StreamCreatedState;
+import de.zalando.aruha.nakadi.service.subscription.zk.ZkSubscriptionClient;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiFunction;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StreamingContext implements SubscriptionStreamer {
+    public final StreamParameters parameters;
+    public final Session session;
+    public final ZkSubscriptionClient zkClient;
+    public final KafkaClient kafkaClient;
+    public final SubscriptionOutput out;
+    public final long kafkaPollTimeout;
+
+    private final ScheduledExecutorService timer;
+    private final BlockingQueue<Runnable> taskQueue = new LinkedBlockingQueue<>();
+    private final BiFunction<Session[], Partition[], Partition[]> rebalancer;
+
+    private State currentState;
+    private ZkSubscriptionClient.ZKSubscription clientListChanges;
+
+    private static final Logger LOG = LoggerFactory.getLogger(StreamingContext.class);
+
+    StreamingContext(
+            final SubscriptionOutput out,
+            final StreamParameters parameters,
+            final Session session,
+            final ScheduledExecutorService timer,
+            final ZkSubscriptionClient zkClient,
+            final KafkaClient kafkaClient,
+            final BiFunction<Session[], Partition[], Partition[]> rebalancer,
+            final long kafkaPollTimeout) {
+        this.out = out;
+        this.parameters = parameters;
+        this.session = session;
+        this.rebalancer = rebalancer;
+        this.timer = timer;
+        this.zkClient = zkClient;
+        this.currentState = new StreamCreatedState(this);
+        this.kafkaClient = kafkaClient;
+        this.kafkaPollTimeout = kafkaPollTimeout;
+    }
+
+    @Override
+    public void stream() throws InterruptedException {
+        // Because all the work is processed inside one thread, there is no need in
+        // additional lock.
+        currentState = new StreamCreatedState(this);
```

Comment:
I think that there's not need to do it again since it was already done in the constructor.